### PR TITLE
added function bytes to Utf8

### LINKED ISSUE #3: Add functionality to convert bytes to utf8 string
Currently library has a function "toHex" to convert typedArray to hexadecimal string. 
There can also be scenarios in which one has to convert typedArray into Unicode(UTF_8). 
add this functionality.

### PR DESCRIPTION
**Pull request by barrytra**
this pull request fixes #3
new function "toUtf8" has been added in both browser.ts and index.ts files. this function efficiently converts a typedArray to Utf8 encoded string.
some new tests have also been added to verify the correctness of the code

### CODE REVIEW

**Comment by barrytra** [comment]
@junderw pls go through it and suggest if some changes are required.

File: ts_src/browser.ts [hunk 1]
```diff
@@ -11,6 +11,10 @@ const HEX_CODEPOINTS: (number | undefined)[] = Array(256)
 const ENCODER = new TextEncoder();
 const DECODER = new TextDecoder("ascii");
 
+export function toUtf8(bytes: Uint8Array): string {
+  return DECODER.decode(bytes);
```
Comment (junderw) [inline ts_src/browser.ts:15]: The `TextDecoder` is not returning utf8 strings.

`Uint8Array.from([ 227, 129, 147, 227, 130, 147, 227, 129, 171, 227, 129, 161, 227, 129, 175 ])`

This should equal `"こんにちは"` but it doesn't. It returns `"ã\x81“ã‚“ã\x81«ã\x81¡ã\x81¯"`

Can you figure out why?

**Review by junderw — CHANGES_REQUESTED** [review]
Errors on a valid UTF8 string in Japanese. See the comment.

File: ts_src/browser.ts (hunk 1 above)
Comment (barrytra) [reply ts_src/browser.ts:15]: okay.. i'll look into it

Comment (barrytra) [reply ts_src/browser.ts:15]: I reviewed it. It turns out that the ```function DECODER.decode(bytes);``` returns correct answer in browser console. However it gives some garbage value in node.js.  But I assume this function which i wrote in browser.ts will only be used in browser and not node. So it should work perfectly fine. And for node.js, I created a different function in Index.ts which returns
```Buffer.from(bytes || []).toString();```
Let me know @junderw if I am getting something wrong here.

**Comment by barrytra** [comment]
The reason "toUtf8" was giving a different output was ```TextDecoder("ascii")``` was converting it to string values whereas it should only be ```textDecoder()``` to convert to utf8 string. another test has been added which was creating problem. This has now been fixed and all the tests run fine

**Review by junderw — APPROVED** [review]

**Pull request merged**